src/modules/dashboard/components/ExpenditurePage/Stages/StageItem.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { MessageDescriptor, useIntl } from 'react-intl';
+import classNames from 'classnames';
+
+import styles from './StageItem.css';
+
+interface Props {
+  label: string | MessageDescriptor;
+  isActive: boolean;
+  isFirst?: boolean;
+}
+
+const StageItem = ({ label, isActive, isFirst }: Props) => {
+  const { formatMessage } = useIntl();
+
+  const labelText = typeof label === 'object' ? formatMessage(label) : label;
```

Comment:
If `label` comes in as `null` it's `typeof` result will still be `object`.

I would actually check for the existence of the `id` prop on `label` and base my logic around that.

```suggestion
  const labelText = label?.id ? formatMessage(label) : label;
```